lib/linux_admin/network_interface/network_interface_rh.rb

```diff
@@ -0,0 +1,138 @@
+require 'ipaddr'
+require 'pathname'
+
+module LinuxAdmin
+  class NetworkInterfaceRH < NetworkInterface
+    IFACE_DIR = "/etc/sysconfig/network-scripts"
+
+    # @return [Hash<String, String>] Key value mappings in the interface file
+    attr_reader :interface_conf
```

Comment:
I could be private, but I don't see an issue with giving someone the chance to see what is actually in the file, this is just exactly the key=>value pairs that will get written so it will make sense to someone who knows what the syntax is for the interface configuration.

I think `:interface_config` is good.  I don't think there is another word to use.